src/mlpack/methods/neighbor_search/knn_main.cpp

```diff
@@ -78,6 +82,9 @@ PARAM_FLAG("single_mode", "If true, single-tree search is used (as opposed to "
     "dual-tree search).", "S");
 PARAM_DOUBLE_IN("epsilon", "If specified, will do approximate nearest neighbor "
     "search with given relative error.", "e", 0);
+PARAM_STRING_IN("effective_error", "If specified, will compare the results "
+    "against the provided distances file, and will print the average relative "
+    "error.", "E", "");
```

Comment:
Great, sounds good to me.  I guess something worth doing might be printing a warning if the user specifies either `-T` or `-D` but is doing a search that gives exact results.  If you like, I can write the code to do that, since you have already made very many changes. :)